### PR TITLE
chore: update typescript to 5.3.3

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

updates typescript to latest. updates eslint to latest. fixes some typescript errors, ignores some typescript errors. suppresses new eslint errors. enables json linting, fixes json linting errors.

# Test Plan

CI passing, builds building

# Changelog

 - Updates typescript to 5.3.3

# Review requests

check local builds

# Risk assessment

medium-high

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -5,6 +5,7 @@
 **/build/**
 **/venv/**
 .opentrons_config
+**/tsconfig*.json
```
Comment (brenthagen) [inline .eslintignore:8]: typescript-eslint/parser seems unable to parse tsconfig json files, so we're ignoring them

File: .eslintrc.js [hunk 2]
```diff
@@ -7,7 +7,12 @@ module.exports = {
     project: require('path').join(__dirname, 'tsconfig-eslint.json'),
   },
 
-  extends: ['standard-with-typescript', 'plugin:react/recommended', 'prettier'],
+  extends: [
+    'standard-with-typescript',
+    'plugin:react/recommended',
+    'prettier',
+    'plugin:json/recommended',
```
Comment (brenthagen) [inline .eslintrc.js:14]: turns out, without extending these rules we weren't actually linting json files in edge. so now we're actually linting json

Comment (brenthagen) [reply .eslintrc.js:14]: downside: linting json via typescript-eslint/parser takes about an extra minute to run

Comment (shlokamin) [reply .eslintrc.js:14]: lol fyi @sfoster1

File: protocol-designer/src/navigation/reducers/index.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ const page: Reducer<Page, any> = handleActions(
   },
   'file-splash'
 )
-// @ts-expect-error(sa, 2021-6-21): TS thinks this will only return false, even though TOGGLE_NEW_PROTOCOL_MODAL might yield true
```
Comment (jerader) [inline protocol-designer/src/navigation/reducers/index.ts:16]: nice!!

**Review by shlokamin — APPROVED** [review]
lgtm. smoke tested PD and stuff still seems to work

**Review by jerader — APPROVED** [review]
woo, thank you Brent! I mainly tested PD and LL. Looks good to me!

File: .eslintrc.js (hunk 2 above)
Comment (brenthagen) [reply .eslintrc.js:14]: twas a breaking change in a previous version of https://github.com/azeemba/eslint-plugin-json?tab=readme-ov-file#eslint-plugin-json

**Review by sfoster1 — APPROVED** [review]
schweet

**Pull request merged**